mlflow/spark/__init__.py

```diff
@@ -94,10 +94,8 @@ def get_default_pip_requirements(is_spark_connect_model=False):
 
     # Strip the suffix from `dev` versions of PySpark, which are not
     # available for installation from Anaconda or PyPI
-    pyspark_extras = ["connect"] if is_spark_connect_model else None
-    pyspark_req = re.sub(
-        r"(\.?)dev.*$", "", _get_pinned_requirement("pyspark", extras=pyspark_extras)
-    )
+    pyspark_req_str = "pyspark[connect]" if is_spark_connect_model else "pyspark"
+    pyspark_req = re.sub(r"(\.?)dev.*$", "", _get_pinned_requirement(pyspark_req_str))
```

Comment:
@harupy I ended up refactoring the `_get_pinned_requirement` function to remove the `extras` param, so I had to edit this callsite as well (looks like it was the only other one using the `extras` param). Let me know if it looks alright!

Reply:
Yes, looks good to me!